pkg/aws/core.go

```diff
@@ -441,7 +441,7 @@ func (d *Driver) GetMachineStatus(_ context.Context, req *driver.GetMachineStatu
 
 	// if SrcAnDstCheckEnabled is false then check attribute on instance and return Uninitialized error if not matching.
 	if providerSpec.SrcAndDstChecksEnabled != nil && !*providerSpec.SrcAndDstChecksEnabled {
-		if *requiredInstance.SourceDestCheck {
+		if requiredInstance.SourceDestCheck != nil && *requiredInstance.SourceDestCheck {
```

Comment:
personally fun of
```
if ptr.Deref(requiredInstance.SourceDestCheck, false)
```

Reply:
Addressed in [1f9f63d](https://github.com/gardener/machine-controller-manager-provider-aws/pull/180/commits/1f9f63d86b55131b46ac4d0195520f80adf29682).

Thanks for the review.